src/key.rs

```diff
@@ -437,6 +437,56 @@ where
     }
 }
 
+pub struct StringDMA<'a> {
+    key: &'a RedisKeyWritable,
+    buffer: &'a mut [u8],
+}
+
+impl<'a> StringDMA<'a> {
+    fn new(key: &'a RedisKeyWritable) -> Result<StringDMA<'a>, RedisError> {
+        let mut length: size_t = 0;
+        let dma = raw::string_dma(key.key_inner, &mut length, raw::KeyMode::WRITE);
+        if dma.is_null() {
+            Err(RedisError::Str("Could not read key"))
+        } else {
+            let buffer = unsafe { std::slice::from_raw_parts_mut(dma.cast::<u8>(), length) };
+            Ok(StringDMA { key, buffer })
+        }
+    }
+
+    pub fn write(&mut self, data: &[u8]) -> Result<&mut Self, RedisError> {
+        if self.buffer.len() != data.len() {
+            if raw::Status::Ok == raw::string_truncate(self.key.key_inner, data.len()) {
+                let mut length: size_t = 0;
+                let dma = raw::string_dma(self.key.key_inner, &mut length, raw::KeyMode::WRITE);
+                self.buffer = unsafe { std::slice::from_raw_parts_mut(dma.cast::<u8>(), length) };
+            } else {
+                return Err(RedisError::Str("Failed to truncate string"));
+            }
+        }
+        self.buffer[..data.len()].copy_from_slice(data);
+        Ok(self)
+    }
+
+    pub fn append(&mut self, data: &[u8]) -> Result<&mut Self, RedisError> {
+        let current_len = self.buffer.len();
+        let new_len = current_len + data.len();
+        if raw::Status::Ok == raw::string_truncate(self.key.key_inner, new_len) {
+            let mut length: size_t = 0;
+            let dma = raw::string_dma(self.key.key_inner, &mut length, raw::KeyMode::WRITE);
+            self.buffer = unsafe { std::slice::from_raw_parts_mut(dma.cast::<u8>(), length) };
+        } else {
+            return Err(RedisError::Str("Failed to truncate string"));
+        }
+        self.buffer[current_len..new_len].copy_from_slice(data);
+        Ok(self)
+    }
+
+    pub fn as_bytes(&self) -> &[u8] {
```

Comment:
We can implement Deref
https://doc.rust-lang.org/std/ops/trait.Deref.html

Reply:
do you need Deref? there's nothing to release here. StringDMA only holds borrows and its life time depends on the KeyHolder lifetime

Reply:
If you implement deref, you do not need to implement is empt ly (for example)

Reply:
is_empty is just for users simplicity